src/test/java/ai/datahunters/md/server/server/photos/UploadPhotosEndpointTest.java

```diff
@@ -56,4 +63,43 @@ public void uploadPhotos() throws IOException {
                 );
 
     }
+
+
+    @Test
+    public void probeArchiveAndExtract() throws IOException, CompressorException {
```

Comment:
First of all this guy deserves a separate file.

Second, there is quite some logic with creating files and getting all the entries. Could you prepare a test (one archive and one compressor would be enough) where you have more than 1 file in the archive? Apart from asserting on file names there you could check the expected contents of the file.

Reply:
Bonus points would go if you would not use any spring annotation in this test. Here you have a nice possibility to prepare a piece of code that does not need to know about existence of spring at all.